.github/actions/pr-sanity-checks/action.yaml

```diff
@@ -0,0 +1,62 @@
+name: 'Pull request sanity checks'
+
+defaults:
+  run:
+    shell: 'bash -Eeuo pipefail {0}'
+
+runs:
+  using: composite
+  steps:
+    - name: Run gitlint
+      shell: bash
+      env:
+        BEFORE_REF: ${{ github.event_name == 'pull_request' && github.event.pull_request.base.sha || github.event.before }}
+        AFTER_REF: ${{ github.sha }}
+      run: |
+        pip install gitlint
+        gitlint --commits $BEFORE_REF..$AFTER_REF --config .github/workflows/.gitlint
+    - name: Run shellcheck
+      shell: bash
+      run: |
+        DEBIAN_FRONTEND=noninteractive apt-get update -qy && apt-get install -qy shellcheck
+        ./scripts/shellcheck.sh
+      # https://github.com/rhysd/actionlint/blob/main/docs/usage.md#use-actionlint-on-github-actions
+    - name: Check workflow files
+      shell: bash
+      run: |
+        bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash)
```

Comment:
maybe use the permalink: https://github.com/rhysd/actionlint/blob/6f31c56453b9c256bfb4940a3d1cf8eb596d26dd/scripts/download-actionlint.bash

Reply:
also pin the binary version:
* download the bash script
* execute `bash download-actionlink.sh 1.6.9`